### PR TITLE
Fix some errors on hooks api page

### PR DESCRIPTION
**Pull request by PhMemmel**
While implementing my first hook I stumbled over these - IMO - errors on the hook API page.

### CODE REVIEW

File: docs/apis/core/hooks/index.md [hunk 1]
```diff
@@ -158,9 +158,9 @@ namespace mod_activity\hook;
 
 #[\core\attribute\label('Hook dispatched at the very end of installation of mod_activity plugin.')]
 #[\core\attribute\tags('installation')]
-class installation_finished implements \core\hook\described_hook {
+final class installation_finished {
     public function __construct(
-        public function string $version,
+        public string $version,
```
Comment (andrewnicols) [inline docs/apis/core/hooks/index.md:163]: Bah - this was mean to be using readonly properties, but I mistyped `function` instead.
```suggestion
        public readonly string $version,
```

File: docs/apis/core/hooks/index.md [hunk 2]
```diff
@@ -174,9 +174,9 @@ class installation_finished implements \core\hook\described_hook {
 <?php
 namespace mod_activity\hook;
 
-class installation_finished implements \core\hook\described_hook {
+final class installation_finished implements \core\hook\described_hook {
     public function __construct(
-        public function string $version,
+        public string $version,
```
Comment (andrewnicols) [inline docs/apis/core/hooks/index.md:179]: Same here

**Review by andrewnicols — CHANGES_REQUESTED** [review]
Thanks @PhMemmel ,

Good catch on those. The `public function` were intended to be `public readonly` and be demonstrating constructor property promotion. D'oh!

**Comment by PhMemmel** [comment]
Requested changes added

**Review by andrewnicols — APPROVED** [review]

**Pull request merged**